transaction/signer/src/types.rs

```diff
@@ -182,6 +182,16 @@ impl AsRef<[u8; 32]> for AccountId {
     }
 }
 
+/// Create [AccountId] object from hex-encoded string
+impl From<String> for AccountId {
+    fn from(value: String) -> Self {
+        let mut byte_array = [0u8; 32];
+        hex::decode_to_slice(value, &mut byte_array)
+            .expect("failed to decode account_id hex string");
```

Comment:
Do we really want to `expect` and panic on failure here? Maybe implementing `TryFrom` and returning a `Result<AccountId, hex::FromHexError>` would be better?

Reply:
definitely better. updated PR to impl TryFrom trait.